zarr/storage.py

```diff
@@ -2255,29 +2253,32 @@ def rmdir(self, path=None):
         dir_path = normalize_storage_path(self._append_path_to_prefix(path))
         if dir_path:
             dir_path += '/'
-        for blob in self.client.list_blobs(self.container, prefix=dir_path):
-            self.client.delete_blob(self.container, blob.name)
-
-    def getsize(self, path=None):
-        from azure.storage.blob import Blob
-        store_path = normalize_storage_path(path)
-        fs_path = self.prefix
-        if store_path:
-            fs_path = self._append_path_to_prefix(store_path)
-        if self.client.exists(self.container, fs_path):
-            return self.client.get_blob_properties(self.container,
-                                                   fs_path).properties.content_length
-        else:
-            size = 0
-            if fs_path == '':
-                fs_path = None
-            else:
-                fs_path += '/'
-            for blob in self.client.list_blobs(self.container, prefix=fs_path,
-                                               delimiter='/'):
-                if type(blob) == Blob:
-                    size += blob.properties.content_length
-            return size
+        for blob in self.client.list_blobs(name_starts_with=dir_path):
+            self.client.delete_blob(blob)
+
+    # It is possible azure.store.blob doesn't provide the content_length attribute on
+    # the blob propoeries object anymore. Something to look into.
+    #
+    # def getsize(self, path=None):
```

Comment:
Flagging this as one major thing I haven't figured out.

Reply:
https://azuresdkdocs.blob.core.windows.net/$web/python/azure-storage-blob/12.5.0/azure.storage.blob.html#azure.storage.blob.BlobProperties